src/engine/Sort.cpp

```diff
@@ -52,30 +53,126 @@ std::string Sort::getDescriptor() const {
 }
 
 // _____________________________________________________________________________
-Result Sort::computeResult([[maybe_unused]] bool requestLaziness) {
-  using std::endl;
-  AD_LOG_DEBUG << "Getting sub-result for Sort result computation..." << endl;
-  std::shared_ptr<const Result> subRes = subtree_->getResult();
+Result Sort::computeResult(bool requestLaziness) {
+  // Check if we should use external sorting.
+  bool useExternalSort =
+      getRuntimeParameter<&RuntimeParameters::sortExternal_>();
+
+  // Get the input lazily if we want to do external sorting, and materialized
+  // otherwise.
+  std::shared_ptr<const Result> subRes = subtree_->getResult(useExternalSort);
```

Comment:
Now here comes the part that is debatable.
I would change this as follows:

```suggestion
  std::shared_ptr<const Result> subRes = subtree_->getResult(requestLaziness && useExternalSort);
```

This would also allow to get rid of the code downstream that does

```cpp
if (!requestLaziness) {
    IdTable result{numColumns, allocator()};
    result.reserve(sorter->size());
    for (const auto& block : sorter->getSortedBlocks<0>()) {
      checkCancellation();
      result.insertAtEnd(block);
    }
    cancellationHandle_->resetWatchDogState();
    checkCancellation();
    return {std::move(result), resultSortedOn(), std::move(mergedLocalVocab)};
  }
```

because at this point it's always requested lazily if external sort is performed.

The practical difference this makes is that for the scenarios in which this `Sort` operation has to produce a fully materialized result that is too big to store in memory, this will fail earlier, before even hitting this operation and sorting everything slowly on disk just to run into memory allocation problems when reading from disk. For all other cases this should make no difference at all.